renovate/src/Renovate/BinaryFormat/ELF/Rewriter/Env.hs

```diff
@@ -0,0 +1,171 @@
+{-
+Module           : Renovate.BinaryFormat.ELF.Rewriter.Env
+Description      : Construct the read-only environment for ELF rewriting
+Copyright        : (c) Galois, Inc 2020
+License          : BSD3
+Maintainer       : Langston Barrett <langston@galois.com>
+Stability        : provisional
+
+This currently consists only of locating a good start address for the additional
+text section.
+-}
+
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE GADTs #-}
+
+module Renovate.BinaryFormat.ELF.Rewriter.Env
+  ( RewriterEnv
+  , makeRewriterEnv
+  , _reSegmentMaximumSize
+  , _reSegmentVirtualAddress
+  ) where
+
+import           GHC.Generics (Generic)
+
+import qualified Control.Monad.Catch as C
+import qualified Control.Monad.Fail as Fail
+import qualified Data.List as L
+import qualified Data.Map as Map
+import qualified Data.Ord as O
+
+import qualified Data.ElfEdit as E
+import qualified Data.Macaw.CFG as MM
+
+import           Renovate.Config
+import qualified Renovate.ISA as RI
+
+import Renovate.BinaryFormat.ELF.Common
+
+-- | Read-only environment for ELF rewriting
+data RewriterEnv arch =
+  RewriterEnv { _reSegmentVirtualAddress :: E.ElfWordType (MM.ArchAddrWidth arch)
+              -- ^ Address of the new text section
+              , _reSegmentMaximumSize :: E.ElfWordType (MM.ArchAddrWidth arch)
```

Comment:
@travitch I'm not 100% sure that this is an accurate name/documentation. Can you double check on that?

Reply:
I think the names are fine.  If we are going to export these, can we export lenses instead of the `_`-prefixed names?  Or if we don't need lenses (it seems like we don't), then simple accessors without the `_`

Reply:
Yeah, I just prefiexed with _ to cause less breakage in case we ever did want lenses, but I can remove them.